war/src/main/scss/modules/dialogs.scss

```diff
@@ -0,0 +1,83 @@
+$jenkins-dialog-padding: 1rem;
+
+.jenkins-dialog {
```

Comment:
I wonder why our dialogs do not have a clear visual separation of the title and the button bar. Everything uses the same color and no line separates these elements. But maybe this is a matter of preference. I prefer the Boostrap styling in this case...

Reply:
Something like this:
<img width="363" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/fdaeac91-0e87-4939-a4c1-5edb341e9eb4">

or this:
<img width="360" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/d59f85a1-094a-4a4d-8697-a57d1ba7a60d">

Reply:
I'm not sold on the buttons in the middle currently, 

see:
https://uxdesign.cc/buttons-placement-and-order-bb1c4abadfcb

which recommends on the right

and on button ordering:
https://ux.stackexchange.com/questions/1072/ok-cancel-on-left-right

I think generally in Jenkins we have confirm on the left and cancel on the right (but may be wrong)

Reply:
(I see you've moved them to the right in later commits than I was testing in the design library pull request)

What do you think about the button order?

Reply:
> Something like this

Yes, I think such a styling will make the different areas more obvious.

Reply:
What we have on config pages is that buttons are on the left and we have first `Save` then `Apply`
Maybe we should also make the button placement depend on the dialog type.
For `form` dialogs it might be better to have the buttons on the left with `Submit` being the first similar to the configure pages. This might also apply to the `prompt` dialog.
For other dialogs we can put them to the right and try to detect the OS and then put `OK` left (windows) or right (Mac, Linux).

Reply:
I'm not a designer, maybe @janfaracik has an idea what looks better?

Reply:
> What we have on config pages is that buttons are on the left and we have first `Save` then `Apply` Maybe we should also make the button placement depend on the dialog type. For `form` dialogs it might be better to have the buttons on the left with `Submit` being the first similar to the configure pages. This might also apply to the `prompt` dialog. For other dialogs we can put them to the right and try to detect the OS and then put `OK` left (windows) or right (Mac, Linux).

This sounds good to me.

Reply:
applied now button placement that way.
Also applied now background color to title and buttons

Reply:
This doesn't look right on dark theme at least:
<img width="342" alt="image" src="https://github.com/jenkinsci/jenkins/assets/21194782/5ebfaefe-0b53-4a85-a948-cbe595c05649">

On light it was pretty much invisible on my mac screen:
<img width="328" alt="image" src="https://github.com/jenkinsci/jenkins/assets/21194782/02b2d780-1481-4249-985b-0630aaf1459c">

Would a line work instead?

Reply:
It feels weird the buttons moving all over the place depending on the dialog type, it may just be a problem in the design library as you see them all soon after each other and e.g. when you use the prompt one and alert pops up after but the buttons are on different side.

what do you think?

-----

Dark theme is looking good to me although the line seems a bit prominent on the default theme:
![image](https://github.com/jenkinsci/jenkins/assets/21194782/db22da68-a0e1-45cf-8bda-f7bfa4905299)

![image](https://github.com/jenkinsci/jenkins/assets/21194782/6cb53aae-d667-4282-8fc1-fb17ce22e3f3)

Reply:
Actually there might be cases where you can see multiple dialogs with different type directly after each other. E.g. in matrix auth, plugin when you add a new user (uses a prompt) and that user is already added you would get an alert.

Reply:
default theme with `var(--medium-grey)`

<img width="248" alt="image" src="https://github.com/jenkinsci/jenkins/assets/17767050/6341d4cc-a921-4a18-959b-aaf5142f8dbb">